backend/src/controllers/comparisons/comparisons.controller.ts

```diff
@@ -54,31 +64,47 @@ const clearComparison = async (
 };
 
 const deleteCarFromComparison = async (
-  req: TypedRequestQuery<{
-    complectationId: string;
-    userId: string;
-  }>,
+  req: TypedRequest<
+    { complectationId: string },
+    { tokenPayload: TokenPayload }
+  >,
   res: Response,
   next: NextFunction,
 ): Promise<void> => {
   try {
-    const comparison = await comparisonsService.deleteCarFromComparison(
-      req.query,
-    );
+    const comparison = await comparisonsService.deleteCarFromComparison({
+      complectationId: req.params.complectationId,
+      userId: req.body.tokenPayload.sub,
+    });
     res.status(httpStatus.OK).json(comparison);
   } catch (error) {
     next(error);
   }
 };
 
 const getActiveComparisonCars = async (
-  req: TypedRequestQuery<{ userId: string }>,
+  req: TypedRequestBody<{ tokenPayload: TokenPayload }>,
   res: Response,
   next: NextFunction,
 ): Promise<void> => {
   try {
     const comparison = await comparisonsService.getActiveComparisonCars(
-      req.query.userId.toString(),
+      req.body.tokenPayload.sub,
+    );
+    res.status(httpStatus.OK).json(comparison);
+  } catch (error) {
+    next(error);
+  }
+};
+
+const getActiveComparisonStatus = async (
+  req: TypedRequestBody<{ tokenPayload: TokenPayload }>,
+  res: Response,
+  next: NextFunction,
+): Promise<void> => {
```

Comment:
Get requests cannot have the `body`. Is the `tokenPayload` injected in the middleware?

Reply:
Yes it is injected

Reply:
Hmm, then it's injected in the wrong place. Because `tokenPayload` should be a part of the request, **_not its body_**. Could create a new type that will include the `tokenPayload` in the request but still would be able to have a generic type for the body/query?

Reply:
Good, then lets defer it for now :ok: